lib/Target/CBackend/CBackend.cpp

```diff
@@ -4643,6 +4644,9 @@ void CWriter::printIntrinsicDefinition(FunctionType *funT, unsigned Opcode,
     case Intrinsic::umin:
       Out << "  r = a < b ? a : b;\n";
       break;
+    case Intrinsic::is_constant:
+      Out << " r = false;\n";
```

Comment:
I really like this solution, I think I considered doing the same myself a while back, but didn't get round to it for whatever reason.

I want to suggest adding a comment to the generated output so it's more obvious what's going on when reading it, i.e. `Out << " r = 0 /* llvm.is.constant */;\n";` perhaps?